webapp/components/MasonryGrid/MasonryGridItem.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <ds-grid-item :rowSpan="rowSpan">
+    <slot></slot>
+  </ds-grid-item>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      rowSpan: 10,
+    }
+  },
+  methods: {
+    calculateItemHeight() {
+      this.$parent.$emit('calculating-item-height')
+      this.$nextTick(() => {
+        const gridStyle = this.$parent.$el.style
+        const rowHeight = parseInt(gridStyle.gridAutoRows)
+        const rowGapValue = gridStyle.rowGap || gridStyle.gridRowGap
+        const rowGap = parseInt(rowGapValue)
+        const itemHeight = this.$el.clientHeight
+
+        this.rowSpan = Math.ceil((itemHeight + rowGap) / (rowHeight + rowGap))
+        this.$parent.$emit('finished-calculating-item-height')
+      })
+    },
+  },
+  mounted() {
+    const image = this.$el.querySelector('img')
+    if (image) {
+      image.onload = () => this.calculateItemHeight()
+    } else {
+      // use timeout to make sure layout is set up before calculation
+      setTimeout(() => this.calculateItemHeight(), 0)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-08-20T07:05:01Z" title="Tuesday, August 20th 2019, 9:05:01 am +02:00">Aug 20, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,38 @@
+<template>
+  <ds-grid-item :rowSpan="rowSpan">
+    <slot></slot>
+  </ds-grid-item>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      rowSpan: 10,
+    }
+  },
+  methods: {
+    calculateItemHeight() {
+      this.$parent.$emit('calculating-item-height')
+      this.$nextTick(() => {
+        const grid = this.$parent
+        const rowHeight = parseInt(grid.$el.style.gridAutoRows)
+        const rowGap = parseInt(grid.$el.style.gridRowGap)
+        const itemHeight = this.$el.clientHeight
+
+        this.rowSpan = Math.ceil((itemHeight + rowGap) / (rowHeight + rowGap))
+        this.$parent.$emit('finished-calculating-item-height')
+      })
+    },
+  },
+  mounted() {
+    const image = this.$el.querySelector('img')
+    if (image) {
+      image.onload = () => this.calculateItemHeight()
+    } else {
+      // use timeout to make sure layout is set up before calculation
+      setTimeout(() => this.calculateItemHeight(), 0)
```


I like that you added a comment to explain this, but if the timeout is `0`, I'm a little confused what this does

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-08-20T09:24:50Z" title="Tuesday, August 20th 2019, 11:24:50 am +02:00">Aug 20, 2019</time>_

---


This has to do with the [JavaScript event loop](https://stackoverflow.com/questions/33955650/what-is-settimeout-doing-when-set-to-0-milliseconds/33955673). The timeout – even though it is set to `0` – sends the execution of this function to the back of the queue.

I am not sure which function _exactly_ we are waiting for because I don't know how Vue.js works its magic in the background but I know that without the timeout the layout is messed up. :smile: